test/pj-rehearse-integration/run.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# This test sets up a temporary testing repository
+# resembling `openshift/release` and tries to execute
+# pj-rehearse as if run over a PR for that repo
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+workdir="$( mktemp -d )"
+trap 'rm -rf "${workdir}"' EXIT
+
+test_root="$( dirname "${BASH_SOURCE[0]}")"
+master_data="${PWD}/${test_root}/master"
+candidate_data="${PWD}/${test_root}/candidate"
+fake_openshift_release="${workdir}/repo"
+expected_rehearsed_jobs="${test_root}/expected_jobs"
+
+echo "[INFO] Preparing fake input repository..."
+mkdir "${fake_openshift_release}"
+pushd "${fake_openshift_release}" >/dev/null
+git init --quiet
+cp -R "${master_data}"/* .
+git add ci-operator cluster
+git commit -m "Master version of openshift/release" --quiet
+base_sha="$(git rev-parse HEAD)"
+cp -R "${candidate_data}"/* .
+git add ci-operator cluster
+git commit -m "Candidate version of openshift/release" --quiet
+candidate_sha="$(git rev-parse HEAD)"
+popd >/dev/null
+
+echo "[INFO] Running pj-rehearse in dry-mode..."
+submitted_yamls="${workdir}/rehearsals.yaml"
+error_log="${workdir}/errors.log"
+export JOB_SPEC='{"type":"presubmit","job":"pull-ci-openshift-release-master-rehearse","buildid":"0","prowjobid":"uuid","refs":{"org":"openshift","repo":"release","base_ref":"master","base_sha":"'${base_sha}'","pulls":[{"number":1234,"author":"petr-muller","sha":"'${candidate_sha}'"}]}}'
+
+if ! pj-rehearse --dry-run=true --no-fail=false --candidate-path "${fake_openshift_release}" > "${submitted_yamls}" 2> "${error_log}"; then
+  echo "[ERROR] pj-rehearse failed:"
+  cat "${error_log}"
+  exit 1
+fi
+
+echo "[INFO] Validating created rehearsals"
+rehearsed_jobs="${workdir}/rehearsed_jobs"
+grep "^  job: rehearse-1234-pull-ci" "${submitted_yamls}" | cut -c '8-' | sort  > "${rehearsed_jobs}"
```

Comment:
While this is like 70% of what we want, could we invent a different type of dry-run mode or something that will either POST to a fake server we have or just spit out the YAML it would have POSTed so we can validate like 100% of the output here?

Reply:
I considered it but decided not to because the lower-level details are tested in lower level tests. But I'm willing to negotiate :)

Dry-run is certainly worth fixing so it outputs a proper YAML (list of jobs instead of concatenated individual job YAMLs) so it can be parsed back - that's quite easy. We could then write a Python helper or something to validate it against some expected output, but we would need to deal with the dynamic fields so it's not ultra trivial, and I'm not sure it's worth it because we test the job->rehearsal translation on unit level.

Reply:
Unit tests can't replace e2e! I think we should try -- dynamic fields can be elided with `diff --ignore-matching-lines`, we manage that with k8s resources often. Plus, if we're not actually talking to a server we won't be getting fields like `creationTimestamp` so we should have fewer dynamic fields.

Reply:
Not replace but complement, which I think is the case here. But OK, I'll give it a shot.